meilisearch-http/src/routes/setting.rs

```diff
@@ -196,17 +198,19 @@ async fn update_rules(
     let index = data
         .db
         .open_index(&path.index_uid)
-        .ok_or(ResponseError::index_not_found(&path.index_uid))?;
+        .ok_or(Error::index_not_found(&path.index_uid))?;
 
     let settings = Settings {
         ranking_rules: Some(body.into_inner()),
         ..Settings::default()
     };
 
-    let mut writer = data.db.update_write_txn()?;
-    let settings = settings.into_update().map_err(ResponseError::bad_request)?;
-    let update_id = index.settings_update(&mut writer, settings)?;
-    writer.commit()?;
+    let settings = settings.into_update().map_err(Error::bad_request)?;
+
+    let update_id = data.db.update_write::<_, _, ResponseError>(|writer| {
+        let update_id = index.settings_update(writer, settings)?;
+        Ok(update_id)
+    })?;
```

Comment:
```suggestion
    let update_id = data.db.update_write::<_, _, ResponseError>(|writer| {
        index.settings_update(writer, settings)
    })?;
```

Reply:
Are you sure this is required to specify the error type here?
```suggestion
    let update_id = data.db.update_write(|writer| index.settings_update(writer, settings))?;
```